general/mem_manager.hpp

```diff
@@ -1007,8 +1007,12 @@ inline void Memory<T>::Delete()
    const bool mt_host = h_mt == MemoryType::HOST;
    const bool std_delete = !registered && mt_host;
 
-   if (std_delete ||
-       MemoryManager::Delete_((void*)h_ptr, h_mt, flags) == MemoryType::HOST)
+   if (!std_delete)
+   {
+      MemoryManager::Delete_((void*)h_ptr, h_mt, flags);
+   }
+
+   if (std_delete || mt_host)
```

Comment:
This is equivalent to just `if (mt_host)` because `if ((A && B) || B)` is the same as `if (B)`.

Reply:
Done.